File: scripts/find-missing-details.mjs
import fs from "node:fs";
import path from "node:path";

const filePath = path.join(process.cwd(), "src", "data", "recipes.ts");
const text = fs.readFileSync(filePath, "utf8");
const detailsBlock = text.split("const recipeDetailsMap: Record<string, RecipeDetailsEntry> = {")[1].split("};\n\nexport const recipeLibrary")[0];
const recipeIds = [...text.matchAll(/id: "([^"]+)"/g)].map((m) => m[1]).filter((id) => !id.startsWith("today-") && !id.startsWith("tomorrow-") && !id.startsWith("day-"));
const detailIds = [...detailsBlock.matchAll(/\n\s*"([^"]+)": \{/g)].map((m) => m[1]);
const missing = recipeIds.filter((id) => !detailIds.includes(id));
console.log(JSON.stringify({ recipeCount: recipeIds.length, detailCount: detailIds.length, missing }, null, 2));

File: scripts/find-missing-details.py
import re
from pathlib import Path

text = Path("src/data/recipes.ts").read_text(encoding="utf-8")
entries = re.findall(r'\{\s*id: "([^"]+)",[\s\S]*?category: "([^"]+)"[\s\S]*?\n  \},', text)
ing = set(re.findall(r'"([^"]+)": \[', text.split('const recipeDetailsMap')[0]))
det = set(re.findall(r'"([^"]+)": \{', text.split('const recipeDetailsMap')[1]))
missing = [(recipe_id, category) for recipe_id, category in entries if recipe_id in ing and recipe_id not in det]
print(f"count {len(missing)}")
for recipe_id, category in missing:
    print(category, recipe_id)
